library/Vanilla/Formatting/Quill/Blots/Lines/HeadingBlot.php

```diff
@@ -41,12 +42,38 @@ public function getGroupClosingTag(): string {
     }
 
     /**
-     * @inheritDoc
+     * Since headings are always one line, they don't need special line starts or ends.
+     *
+     * The group tags are enough.
+     */
+    public function renderLineStart(): string {
+        return "";
+    }
+
+    /**
+     * Since headings are always one line, they don't need special line starts or ends.
+     *
+     * The group tags are enough.
      */
+    public function renderLineEnd(): string {
+        return "";
+    }
+
     public function isOwnGroup(): bool {
         return true;
     }
 
+    /**
+     * The heading blot can be the ONLY overriding blot in a group. Even other headings.
+     *
+     * @param BlotGroup $group
+     * @return bool
+     */
+    public function shouldClearCurrentGroup(BlotGroup $group): bool {
+        $overridingBlot = $group->getPrimaryBlot();
+        return !!$overridingBlot;
```

Comment:
This is technically fine, but I think you could get a more explicit result out of using something like [`boolval`](http://php.net/boolval) or just [casting the type to a boolean](http://php.net/manual/en/language.types.type-juggling.php#language.types.typecasting).